src/libpmemset/part.c

```diff
@@ -132,9 +134,33 @@ pmemset_part_map_iterate(struct pmemset_part_map *pmap, size_t offset,
  */
 static int
 pmemset_pmem2_map_delete_cb(struct pmemset_part_map *pmap,
-		struct pmem2_map *map, void *arg)
+		struct pmem2_map *p2map, void *arg)
 {
-	return pmem2_map_delete(&map);
+	struct pmemset_sds_record *sds_record = pmemset_sds_find_record(p2map);
```

Comment:
do you really need all of these functions to be in the external interface for sds? can't you just have "delete sds for this map" function?